lib/interceptors/csrf-token.ts

```diff
@@ -0,0 +1,29 @@
+import { generateUrl } from '@nextcloud/router'
+
+const RETRY_KEY = Symbol('csrf-retry')
+
+export const onError = axios => async (error) => {
+	const { config, response, request: { responseURL } } = error
+	const { status } = response
+
+	if (status === 412
+		&& response?.data?.message === 'CSRF check failed'
```

Comment:
The only way to test this is with the message?
I'm afraid about that fails with any small change from server, like case, translations...


If not, we can try to turn it more resilient:

```suggestion
		&& response?.data?.message?.toLowerCase() === 'csrf check failed'
```

Reply:
Checking status code and message isn't bullet proof but good enough. The worst that can happen is that the message changes one day and the request is no longer retried. Then we can adjust this library. But this won't break anything.